rust/lance-encoding/src/testing.rs

```diff
@@ -417,6 +420,159 @@ impl TestCases {
+            ArrayEncodingEnum::Block(block) => {
+                chain.push("block".to_string());
+                // The block scheme (e.g., "lz4", "zstd") is stored in block.scheme
+                if !block.scheme.is_empty() {
+                    chain.push(block.scheme.clone());
+                }
+            }
+            ArrayEncodingEnum::Rle(_) => chain.push("rle".to_string()),
+            ArrayEncodingEnum::PackedStruct(_) => chain.push("packed_struct".to_string()),
+            ArrayEncodingEnum::PackedStructFixedWidthMiniBlock(_) => {
+                chain.push("packed_struct_fixed_width".to_string())
+            }
+            ArrayEncodingEnum::Nullable(nullable) => {
+                chain.push("nullable".to_string());
+                if let Some(ref nullability) = nullable.nullability {
+                    match nullability {
+                        pb::nullable::Nullability::NoNulls(no_nulls) => {
+                            if let Some(ref values) = no_nulls.values {
+                                chain.extend(extract_array_encoding_chain(values));
+                            }
+                        }
+                        pb::nullable::Nullability::SomeNulls(some_nulls) => {
+                            if let Some(ref values) = some_nulls.values {
+                                chain.extend(extract_array_encoding_chain(values));
+                            }
+                        }
+                        pb::nullable::Nullability::AllNulls(_) => {
+                            chain.push("all_nulls".to_string());
+                        }
+                    }
+                }
+            }
+            ArrayEncodingEnum::FixedSizeList(fsl) => {
+                chain.push("fixed_size_list".to_string());
+                // items is a repeated field in FixedSizeList
+                if let Some(items) = &fsl.items {
+                    chain.extend(extract_array_encoding_chain(items));
+                }
+            }
+            ArrayEncodingEnum::List(_) => {
+                chain.push("list".to_string());
+                // List doesn't have an items field in the simple message
+            }
+            ArrayEncodingEnum::Struct(_) => {
+                chain.push("struct".to_string());
+                // SimpleStruct doesn't have fields array
+            }
+            ArrayEncodingEnum::GeneralMiniBlock(general) => {
+                chain.push("general_miniblock".to_string());
+                // Check the inner encoding
+                if let Some(ref inner) = general.inner {
+                    let inner_chain = extract_array_encoding_chain(inner);
+                    chain.extend(inner_chain);
+                }
+            }
+            ArrayEncodingEnum::ByteStreamSplit(_) => {
+                chain.push("byte_stream_split".to_string());
+            }
+        }
+    }
+
+    chain
+}
```

Comment:
It would be nice if there was some way to generalize this but I don't think protobuf lends itself to that very nicely :frowning:

Reply:
Let me give this a try.